### ISSUE TITLE
[Feature] 3rd exit code to get my pipelines yellow

### ISSUE THREAD

**Issue by janitor-of-code** [zavoloklom/docker-compose-linter#309]
Howdy!  

I'd like this tool to have an exit code other than `0` and `1` i.e. success and failure.  
For findings that aren't errors but still need to be addressed.  

The specific use case is that we are running dc-lint in a github pipeline and my colleagues mostly just watch for the color of icons: green - yellow - red and sometimes don't catch the `Code Quality scans found 1 new finding`-message.   

## Proposed Solution

Starting from the existing exit logic in `cli.ts` line 140:  
Adding another `elif` and check for message length:  

```ts
  if (totalErrors > 0) {
    logger.debug(LogSource.CLI, `${totalErrors} errors found`);
    process.exit(1);
  } else if (cliArguments.maxWarnings >= 0 && totalWarnings > cliArguments.maxWarnings) {
    logger.debug(
      LogSource.CLI,
      `Warning threshold exceeded: ${totalWarnings} warnings (max allowed: ${cliArguments.maxWarnings})`,
    );
    process.exit(1);
  } else if (lintResults.messages.length > 0) {
    logger.debug(LogSource.CLI, `Non-zero exit on findings`);
    process.exit(2);
  }

  logger.debug(LogSource.CLI, 'All files are valid.');
  process.exit(0);
};
```

I'm not sure if a cli argument should be used (and how that should be named) to trigger that behavior.  
I also don't know which number makes the most sense and picked `2` because yaml-lint uses that.  

## Alternatives Considered

I went ahead and added some bash to my pipes that watches the output file:  

```yaml
.compose-lint:
  image:
    name: zavoloklom/dclint:3.1.0-alpine
    entrypoint: ['']
  script:
    - >
      /bin/dclint . -r -c "${LINT_RULES_PATH}"
      -f codeclimate -o gl-codequality.json
    - "[ $(cat gl-codequality.json | wc -c) -le 4 ] || (exit 137)"
  artifacts:
    reports:
      codequality: ./gl-codequality.json
  stage: lint
  allow_failure:
    exit_codes: 137
```

This does the trick for now for me:
There's a yellow exclamation symbol on my pipes
